Improved custom subscription docs (#149)
graphile/graphile.github.io: src/pages/postgraphile/subscriptions.md

_This feature requires PostGraphile v4.4.0 or higher._

### Introduction

Pass `--subscriptions` (or `subscriptions: true`) to PostGraphile and we'll
enhance GraphiQL with subscription capabilities and give your PostGraphile
server the power of websocket communications. This will enable the websocket endpoint.

> Additionally, you'll have to add the [`@graphile/pg-pubsub` plugin](https://www.npmjs.com/package/@graphile/pg-pubsub). It's intended that you use this plugin as a provider of realtime data to other plugins which can use it to add subscription fields to your API. For example, this plugin will add the `@pgSubscriptions` directive to easily define your own subscriptions using LISTEN/NOTIFY with makeExtendSchemaPlugin; and adds the --simple-subscriptions feature which, when enabled, adds a simple listen subscription field to your GraphQL API. See below how to enable the plugin for each approach

Reviewer: ```suggestion
Although you can now use `makeExtendSchemaPlugin` to add your own subscription fields using your own realtime events, it's likely that you'll want to add the [`@graphile/pg-pubsub` realtime provider plugin](https://www.npmjs.com/package/@graphile/pg-pubsub) so that you can leverage PostgreSQL's built-in pubsub support. For example, this plugin will allow `makeExtendSchemaPlugin` to use the `@pgSubscriptions` directive to easily define your own subscriptions using PostgreSQL's `LISTEN`/`NOTIFY` (recommended for production). This plugin also adds the `--simple-subscriptions` flag that can be used to add a simple listen subscription field to your GraphQL API (useful for experimentation). See below how to enable the plugin for each approach.
```

Author: great

> Finally, you'll notice that your

Reviewer: ```suggestion
If you just use the `--subscriptions` flag alone, you'll notice that your
```

Author: LGTM

> schema still only has `query` and `mutation` types. To add subscriptions to

Reviewer: ```suggestion
schema still only has `query` and `mutation` operations. To add subscriptions to
```

Author: why `operations`? AFAIK Query and Mutation are GraphQL types (special because they are entry points, but types).

Just curious, I don't have a strong disagreement with this one, any sensible reader will understand both options

Reviewer: Sure - fair question! (Edited above to be "operation types" rather than "operations".)

---

`query` and `mutation` are "operation types" defined in the GraphQL spec:

https://graphql.github.io/graphql-spec/June2018/#sec-Language.Operations

These terms do not change.

---

`Query` and `Mutation` are "types" (note the capital letters) and their names can change (in PostGraphile this would be due to inflector plugins). For example in the StarWars API example the `query` operation type is named `Root`:

https://graphql.org/swapi-graphql/

---

I had the option of either capitalising `query/mutation` to `Query/Mutation` or inserting the word `operation` to make the text more accurate. I decided that the `operation` keyword would be more stable, so went that way :+1:

Author: Awesome! thanks for the detailed answer

your GraphQL schema you'll need a plugin to provide the relevant

> `subscription` fields (by extending the `Subscription` type)- or you can write your own [with

Reviewer: ```suggestion
`subscription` fields (by extending the `Subscription` type) - or you can write your own [with
```

  -c mydb
```

#### Enabling with an Express app

When using PostGraphile as a library, you may enable Custom Subscriptions by

> passing the `pluginHook` with the `@graphile/pg-pubsub` plugin, setting `subscriptions:true`

Reviewer: ```suggestion
passing the `pluginHook` with the `@graphile/pg-pubsub` plugin, setting `subscriptions: true`
```

const postgraphileOptions = {
  pluginHook, // add the plugin hook. This will make the @pgSubscription avaiable in our schema definitions
  subscriptions: true, // start the websocket server
  websocketMiddlewares: [
    // Add whatever middlewares you need here, note that they should only
    // manipulate properties on req/res, they must not sent response data. e.g.:
    //
    //   require('express-session')(),
    //   require('passport').initialize(),
    //   require('passport').session(),
  ],
};

const app = express();
app.use(postgraphile(databaseUrl, "app_public", postgraphileOptions));
app.listen(parseInt(process.env.PORT, 10) || 3000);
```

#### Testing your subscription with GraphiQL/GraphQL Playground
To test your subscription you will need to first subscribe and then trigger it. 

To subscribe, in one GraphiQL tab execute
```gql
subscription MySubscription {
  currentUserUpdated {	
    user
    event
  }
}
```
You should get the answer: `"Waiting for subscription to yield data…"`

To trigger the subscription, *in another GraphiQL tab* run a mutation that changes the user. This
will depend on your implementation, for example:
```gql
mutation MyMutation {

>   updateUserById(input: {userPatch: {name: 'foo'}, id: ""}) {

Reviewer: ```suggestion
  updateUserById(input: {userPatch: {name: "foo"}, id: 27}) {
```

Author: hehehe thanks

    clientMutationId
  }
}
```
In this tab you will get the regular mutation answer. Going back to the previous tab, 

> you will see the subscription paylod. You are good to go! This should

Reviewer: ```suggestion
you will see the subscription payload. You are good to go! This should
```
